library/Frontend/src/Juvix/Frontend/Parser.hs

```diff
@@ -146,10 +144,6 @@ expressionGen p =
 expression' :: Parser Types.Expression
 expression' = expressionGen app''
 
--- used to remove app from parsing
```

Comment:
How are the changes in this file related to the new backend? I think these could better be dealt with in their own issue/PR.

Reply:
It's just cleaning up warnings that are clutter. Having a separate PR for just removing warnings is an overkill in my opinion.

Reply:
I was told early that refactoring is something that happens along the way as a means to achieve something, not a goal (PR in this case) in itself. In this case, these changes are not even distracting